### DISCUSSION TITLE
Why pgagroal and not pgBouncer?

### DISCUSSION THREAD

**Discussion by laurenz** [pgagroal/pgagroal#326] [General]
I know the title sounds like I am trolling &ndash; I am not.

After reading through the feature list, I thought: that sounds just like pgBouncer.

So I am curious: what are the features that make pgagroal better than pgBouncer?
It might be a good idea to emphasize these in the documentation.

**Comment by jesperpedersen** [comment]
There are pros/cons of using connection pools.

For PostgreSQL there are 4 major (native) connection pool implementations

1. pgagroal
2. pgbouncer
3. pgpool-II
4. Odyssey

which all share a base feature-set namely authenticating a connection and keeping it in a pool after use.

pgagroal and pgpool-II uses a process based approach for this, and pgbouncer and Odyssey uses a thread based approach.

I don't really want to create a complete feature matrix between the implementations because each have something unique to them.

Our deployments have shown a performance increase in benchmarks, separate passwords for applications and backend connections have been of benefit, and having a Prometheus endpoint has proved to have simplified the setup.

I would say that all users should evaluate their requirements and select the connection pool which matches that the closest.

And, yes - documentation could use a lot of updates - feel free to suggest tutorials on various subjects.
